### PR TITLE
Testing/appv2

### PR DESCRIPTION
**Pull request by attom2**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the new behavior?

- Adds test for users to navigate from /game to /pokedex
- Fixes act() error by switching all tests to async

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Components/App/App.test.js [hunk 1]
```diff
@@ -1,16 +1,13 @@
 import React from 'react';
-import { render, fireEvent, waitFor, act } from '@testing-library/react';
+import { render, fireEvent, waitFor } from '@testing-library/react';
 import App from './App';
 import '@testing-library/jest-dom';
 import { MemoryRouter } from 'react-router-dom';
 import { getAllPokemon, getSinglePokemon } from '../../ApiCalls'
-import { singlePokemonMockData} from './Pokemon-Mock-Data';
```
Comment (attom2) [inline src/Components/App/App.test.js:7]: removes large single pokemon data

File: src/Components/App/App.test.js [hunk 2]
```diff
@@ -29,26 +26,35 @@ describe('App', () => {
         "url": "https://pokeapi.co/api/v2/pokemon/4/"
       },
       {
-        "name": "charmeleon",
+        "name": "bulbasaur",
         "url": "https://pokeapi.co/api/v2/pokemon/5/"
   }])
 
-  getSinglePokemon.mockResolvedValue(singlePokemonMockData)
+  getSinglePokemon.mockResolvedValue({
```
Comment (attom2) [inline src/Components/App/App.test.js:33]: uses only info we are needed to mock getSinglePokemon

Comment (attom2) [reply src/Components/App/App.test.js:33]: charmander is effectively our mock winner

File: src/Components/App/App.test.js [hunk 3]
```diff
@@ -29,26 +26,35 @@ describe('App', () => {
...
-    expect(getByRole('heading', { name: "Who's That Pokémon?" })).toBeInTheDocument();
+
+    const title = await waitFor(() => getByRole('heading', { name: "Who's That Pokémon?" }));
+    expect(title).toBeInTheDocument();
 
   });
 
 
   it('should be able to navigate to /game', async () => {
-    const { debug, getByRole, getByText, getAllByRole, getByPlaceholderText } = render(
```
Comment (attom2) [inline src/Components/App/App.test.js:51]: remove extra variables

File: src/Components/App/App.test.js [hunk 4]
```diff
@@ -59,24 +65,63 @@ describe('App', () => {
...
+    fireEvent.click(gameNav);
     const winningStreak = await waitFor(() => getByRole('heading', {name: "Winning streak: 0"}));
-    debug();
     expect(winningStreak).toBeInTheDocument();
 
-    //Still need to check if winning button (charmander) is clicked, should add 1 to win counter
+  
+    const winningBtn = await waitFor(() => getByRole('button', {name: "charmander"}));
+    fireEvent.click(winningBtn);
+    const winningStreak2 = await waitFor(() => getByRole('heading', { name: "Winning streak: 1" }));
```
Comment (attom2) [inline src/Components/App/App.test.js:94]: when you click winning button, winning streak is now 1

Comment (emontealvo) [reply src/Components/App/App.test.js:94]: Solid integration test. Could we take it a step further and check what happens when a streak ends?

**Comment by emontealvo** [comment]
Nice work! Our test coverage is looking solid/

**Comment by MarchCorbin** [comment]
Nice more efficient for testing now!

**Pull request merged**